### DISCUSSION TITLE
Can't read from S210 Drive

### DISCUSSION THREAD

**Discussion by TimDMottram** [gijzelaerr/python-snap7#499] [Q&A] [answered]
Hi, we are trying out Snap7 (python) to get trace data from our drives (1 x G120, 2 x S210). 

For the G120 everything works like a charm - we get all the data, nice and fast (not fully checked for continuity yet).

For the S210 nothing we try works, we can't read / write a single Parameter, we keep getting 'CPU Refused' error. 

The S210 is slightly different in that it uses different 'Drive Objects' for CU and Application spaces - but as we can't read/write anything at all, I'm a little lost. 

Are there any settings that you think might be set that are stopping the S210 responding? (TIA configuration etc)

The connections seems to open find, lights are on but nobody's home...

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. S210 drives may not support the S7comm protocol. Consider using PROFINET or OPC UA for drive communication.
